account.rs/src/wasm/src/account.rs

```diff
@@ -342,6 +345,55 @@ impl Account {
             .map_err(AccountError::AesGcmSiv)
     }
 
+    #[wasm_bindgen(js_name = decryptNaclUtxo)]
+    pub fn decrypt_nacl_utxo(
+        &self,
+        ciphertext: Vec<u8>,
+        commitment: Vec<u8>,
+    ) -> Result<Vec<u8>, AccountError> {
+        let nonce = GenericArray::from_slice(&commitment[0..24]);
+        let signer_pub_key = SecretKey::from(SECRET_KEY).public_key();
+        let private_key = SecretKey::from(self.encryption_private_key);
+        let nacl_box = Box::new(&signer_pub_key, &private_key);
+        let decrypted_message = nacl_box
+            .decrypt(nonce, &ciphertext[..])
+            .map_err(|_| AccountError::Generic("can't decrypt nacl".to_string()))?;
+
+        Ok(decrypted_message)
+    }
+
+    /**
+     * Encrypts utxo bytes to public key using a nonce and a standardized secret for hmac.
+     * @static
+     * @param publicKey - The public key to encrypt to.
+     * @param bytes_message - The message to be encrypted.
+     * @param commitment - The commitment used to generate the nonce.
+     * @returns The encrypted Uint8Array.
+     */
+    #[wasm_bindgen(js_name = encryptNaclUtxo)]
+    pub fn encrypt_nacl_utxo(
+        public_key: Vec<u8>,
+        message: Vec<u8>,
+        commitment: Vec<u8>,
+    ) -> Result<Vec<u8>, AccountError> {
+        let pub_key_array: [u8; 32] = vec_to_key(&public_key)?;
+        let pub_key = PublicKey::from(pub_key_array);
+        let nonce = GenericArray::from_slice(&commitment[0..24]);
+
+        // CONSTANT_SECRET_AUTHKEY is used to minimize the number of bytes sent to the blockchain.
+        // This results in poly135 being useless since the CONSTANT_SECRET_AUTHKEY is public.
+        // However, ciphertext integrity is guaranteed since a hash of the ciphertext is included in a zero-knowledge proof.
+        let private_key = SecretKey::from(SECRET_KEY);
+
+        let nacl_box = Box::new(&pub_key, &private_key);
+
+        let encrypted_message = nacl_box
+            .encrypt(nonce, &message[..])
+            .map_err(|_| AccountError::Generic("can't encrypt nacl".to_string()))?;
```

Comment:
Is there a way to propagate the actual error from the crate?
Maybe something like this works:
```suggestion
            .map_err(|e| AccountError::Generic(format!("can't encrypt nacl. Error: {}", e.to_string())))?;
```

Reply:
probably makes sense to use this-error as @vadorovsky suggested.